bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -44,30 +50,38 @@
 {
   private final int MAX_COUNT_OF_INMEM_BLOCKS;
   protected final String identifier;
-  private final Integer blocksize;
+  private final int blockSize;
   private HashMap<BitVector, HashSet<DataListener>> listeners = new HashMap<BitVector, HashSet<DataListener>>();
   protected HashSet<DataListener> all_listeners = new HashSet<DataListener>();
   protected Block first;
   protected Block last;
   protected Storage storage;
-  protected ExecutorService autoflushExecutor;
+  protected ExecutorService autoFlushExecutor;
   protected ExecutorService storageExecutor;
+  protected final BlockingQueue<byte[]> freeBuffers;
+  protected final List<AbstractClient> suspendedClients;
+  private volatile int numberOfInMemBlockPermits;
 
   public int getBlockSize()
   {
-    return blocksize;
+    return blockSize;
   }
 
-  public void rewind(int baseSeconds, int windowId) throws IOException
+  public synchronized void rewind(int baseSeconds, int windowId) throws IOException
```

Comment:
Then may be we should make just that code block synchronized. In another comment elsewhere, I mentioned that it's probably not even important that the code block execute. So may be we should wrap it in synchronized or remote it completely.